### DISCUSSION TITLE
How to add a item with a variant in cart using swell-node?

### DISCUSSION THREAD

**Discussion by flayks** [swellstores/community#292] [Help] [answered]
I'm struggling to add an item to a cart with a variant. How could I achieve that using swell-node? I'm using it over swell-js for performance issues as the front library is quite heavy (160kb) and I don't need all the features.

So far I've tried these two options (and many other things), but no luck. It just adds the product to the cart without any variant, or doesn't add it at all 🤷 
```ts
await swell.put('/carts/{id}', {
    id: cart.id,
    items: {
        // Add new item to existing ones
        $set: [
            ...cart.items,
            {
                product_id: productId,
                variant_id: '645be3b1e32f7d1d1b4b83ba',
                variant: {
                    id: '645be3b1e32f7d1d1b4b83ba',
                },
                quantity,
            }
        ]
    },
})
```

What is supposed to be the syntax here? Also how would that work with potentially 2 or 3 variants for the same product item?

**Comment by flayks** [comment]
I think I found a lead that seems to work! From the `swell-js` lib docs:
```ts
await swell.cart.addItem({
  product_id: '5c15505200c7d14d851e510f',
  quantity: 1,
  options: [
    { name: 'Size', value: 'S' },
    { name: 'Color', value: 'Midnight blue' }
  ]
})
```

So directly applied to `swell-node`, it would be:
```ts
await swell.put('/carts/{id}', {
    id: cart.id,
    items: {
        // Add new item to existing ones
        $set: [
            ...cart.items,
            {
                product_id: productId,
                // variants is passed to my function as [{ size: 'Medium' }, { ... }]
                options: variants,
                quantity,
            }
        ]
    },
})
```

If I had a feedback guys, is that it could be much more clear on the docs with clearer examples, because [that is not very obvious](https://developers.swell.is/backend-api/carts/update-a-cart#update-a-cart):

![CleanShot 2023-06-30 at 20 50 29@2x](https://github.com/swellstores/community/assets/273716/9b1629b0-d3ea-4351-ac0f-bbf9195c363e)

**Reply by sags95** [reply, marked as answer]
That's correct you would be passing the options rather than explicitly setting the `variant_id` in most cases. I'll make a note to clarify this in the docs.
